pkg/environment/apply.go

```diff
@@ -182,25 +182,68 @@ func (a *Apply) ApplyBatch() error {
 // get the latest changes (In case any PRs were merged since the pipeline started), and perform
 // the apply of that namespace
 func (a *Apply) applyNamespaceDirs(chunkFolder []string) error {
-	for _, folder := range chunkFolder {
+	erroredNs := []string{}
 
-		// split the path to get the namespace name
-		namespace := strings.Split(folder, "/")
-		a.Options.Namespace = namespace[2]
+	err := util.GetLatestGitPull()
+	if err != nil {
+		return err
+	}
 
-		err := util.GetLatestGitPull()
-		if err != nil {
-			return err
-		}
-		err = a.applyNamespace()
-		if err != nil {
-			return err
-		}
+	done := make(chan bool)
+	defer close(done)
+
+	chunkStream := util.Generator(done, chunkFolder...)
 
+	routineResults := a.parallelApplyNamespace(done, chunkStream)
+
+	results := util.FanIn(done, routineResults...)
+
+	for res := range results {
+		erroredNs = append(erroredNs, res)
 	}
+
+	fmt.Printf("\nerrored ns: %v\n", erroredNs)
 	return nil
 }
 
+func (a *Apply) parallelApplyNamespace(done <-chan bool, dirStream <-chan string) []<-chan string {
+	numRoutines := runtime.NumCPU()
```

Comment:
run as many go routines as we have cores, this proves to be a conservative number of routines because of how lightweight and cheap they are. We should monitor performance in concourse and adjust here if necessary.